circleci/docker-publish

```diff
@@ -26,11 +40,19 @@ REPO=$CIRCLE_PROJECT_REPONAME
 : ${CIRCLE_SHA1?"Missing required env var"}
 SHORT_SHA=${CIRCLE_SHA1:0:7}
 
-echo "Logging into DockerHub..."
-docker login -u $DOCKER_USER -p $DOCKER_PASS --email="$DOCKER_EMAIL"
-
 echo "Building docker image..."
 docker build -t $ORG/$REPO:$SHORT_SHA .
 
-echo "Pushing docker image..."
+echo "Logging into DockerHub..."
+docker login -u $DOCKER_USER -p $DOCKER_PASS --email="$DOCKER_EMAIL"
```

Comment:
The order change I made was to build the image first and then login with a final step of push. 

Build is common to both ecr and dockerhub so is done first. Maybe I should pull dockerhub into its own function